src/DependencyInjection/NeustaConverterExtension.php

```diff
@@ -34,6 +35,13 @@ public function loadInternal(array $config, ContainerBuilder $container): void
      */
     private function registerConverterConfiguration(string $id, array $config, ContainerBuilder $container): void
     {
+        if (!$targetFactoryId = $config['target_factory'] ?? null) {
+            $container->register($targetFactoryId = "{$id}.target_factory", GenericTargetFactory::class)
+                ->setArguments([
+                    '$type' => new Reference($config['target']),
+                ]);
+        }
+
```

Comment:
I'm already trying to separate that into different strategy handlers - otherwise this configuration handling is not readable anymore.
Especially this ternary-operator-not-null-game is here very confusing.

Reply:
Please don't put too much energy into it! These files tend to look very ugly and usually do not conform to normal coding standards. Take a look at the Symfony files, for example: https://github.com/symfony/symfony/blob/6.3/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

I've also never seen them split into separate files, so please let's stick to how Symfony and the other bundles do it.

Reply:
@mike4git what about writing it like this? Would that be an improvement for you?

```suggestion
        $targetFactoryId = $config['target_factory'] ?? "{$id}.target_factory";
        
        if (!isset($config['target_factory'])) {
            $container->register($targetFactoryId, GenericTargetFactory::class)
                ->setArguments([
                    '$type' => new Reference($config['target']),
                ]);
        }

```